src/tests/Microsoft.PowerFx.Interpreter.Tests/ThreadingTest.cs

```diff
@@ -0,0 +1,214 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using Microsoft.PowerFx.Core;
+using Microsoft.PowerFx.Core.Public;
+using Microsoft.PowerFx.Core.Utils;
+using Xunit;
+
+namespace Microsoft.PowerFx.Tests
+{
+    // Do static analysis to look for potential threading issues. 
+    public class ThreadingTest
+    {
+        [Fact]
+        public void CheckIntepreter()
+        {
+            var asm = typeof(RecalcEngine).Assembly;            
+            CheckStatics(asm);
+        }
+
+        [Fact]
+        public void CheckFxCore()
```

Comment:
very minor organizational nit: Tests that cover Core should generally live in Core.Tests (the Interpreter.Tests depends on Core.Tests and has InternalsVisibleTo, so it could still have it's own test that calls the same CheckStatics method)